<?xml version="1.0"?>
<launch>
  <arg name="name" default="eff"/>
  <arg name="number" default="1" />
  <arg name="frame" default="$(arg name)/base_link" />
  <arg name="joy_dev" default="/dev/input/js0" />
  <arg name="x" default="0" />
  <arg name="y" default="0" />
  <arg name="z" default="0" />
  <arg name="path" default="bleh"/>


  <!--remap from="/Robot_1/ground_pose" to="/$(arg name)/Robot_1/ground_pose"/-->
  <remap from="$(arg name)/wheel_control/goal_point" to="wheel_command/goal_point"/>



  <group ns="$(arg name)" >
    <node name="pose_converter" type="pose_converter.py" pkg="crazyflie_tools" args="$(arg name) $(arg number)">
    </node>
    <include file="$(find crazyflie_driver)/launch/crazyflie_add.launch">
      <arg name="name" value="$(arg name)" />
    </include>
    <node name="wheel_control" pkg="crazyflie_teleop" type="wheel_controller.py" output="screen" args="$(arg number)"/>
    <node name="wheel_command" pkg="crazyflie_teleop" type="wheel_command.py" output="screen" args="$(find crazyflie_teleop)/drive_paths/$(arg path) $(arg number)"/>
    <node name="path_display" pkg="crazyflie_tools" type="path_display.py" output="screen" args="$(arg number)">
    </node>
<!--     <include file="$(find crazyflie_teleop)/launch/localization.launch">
      <arg name="name" value="$(arg name)"/>
    </include> -->
  </group>


  <!-- run rviz -->
  <param name="robot_description" command="$(find xacro)/xacro.py $(find crazyflie_description)/urdf/crazyflie.urdf.xacro" />

  <param name="/cf_list/$(arg name)" type="boolean" value="True"/> 

  <node name="cf_selector" pkg="crazyflie_teleop" type="cf_selector.py">
  </node>
  
</launch>
